File: day01/calorie_counting.py
from sys import argv

calories = []
with open(argv[1]) as f:
    elf = 0
    for line in f:
        if line := line.rstrip():
            elf += int(line)
        else:
            calories.append(elf)
            elf = 0

calories.append(elf)
print('Part 1:', max(calories))
print('Part 2:', sum(sorted(calories)[-3:]))

File: day01/calorie_counting.cpp
#include <stdlib.h>
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

int main(int argc, char *argv[]) {
    const string fname = argv[1];

    ifstream infile(fname);
    if (!infile) {
        perror(fname.c_str());
        exit(1);
    }

    vector<int> calories;
    int elf = 0;
    string line;
    while (!infile.eof()) {
        getline(infile, line);
        if (line == "") {
            calories.push_back(elf);
            elf = 0;
        } else {
            elf += stoi(line);
        }
    }
    
    sort(calories.begin(), calories.end());
    cout << "Part 1: " << calories.back() << endl;

    int sum = 0;
    for (size_t i = calories.size() - 3; i < calories.size(); i++)
        sum += calories[i];
    cout << "Part 2: " << sum << endl;
}

